src/v/datalake/arrow_writing_consumer.h

```diff
@@ -0,0 +1,50 @@
+#include "cluster/partition.h"
+#include "model/fundamental.h"
+#include "model/record.h"
+#include "storage/log.h"
+#include "storage/parser_utils.h"
+
+#include <seastar/core/shared_ptr.hh>
+
+#include <arrow/api.h>
+#include <arrow/chunked_array.h>
+#include <arrow/io/api.h>
+#include <arrow/result.h>
+#include <arrow/type_fwd.h>
+#include <parquet/arrow/writer.h>
+#include <ssx/thread_worker.h>
+
+namespace datalake {
+class arrow_writing_consumer {
+    /** Consumes logs and writes the results out to a Parquet file.
+     *
+     * Uses an arrow ChunkedArray with one chunk per batch to avoid large
+     * allocations. However, it does hold all the data in RAM until the file
+     * is written.
+     */
+public:
+    explicit arrow_writing_consumer();
+    ss::future<ss::stop_iteration> operator()(model::record_batch batch);
+    ss::future<std::shared_ptr<arrow::Table>> end_of_stream();
+    std::shared_ptr<arrow::Table> get_table();
```

Comment:
Can you add some comments to these declarations, so users have a rough idea of the workflow for the public API?